language/diem-tools/writeset-transaction-generator/src/main.rs

```diff
@@ -75,15 +78,27 @@ enum Command {
 
 fn save_bytes(bytes: Vec<u8>, path: PathBuf) -> Result<()> {
     std::fs::write(path.as_path(), bytes.as_slice())
-        .map_err(|_| format_err!("Unable to write to path"))
+        .map_err(|err| format_err!("Unable to write to path: {:?}", err))
 }
 
-fn diem_framework_modules() -> Vec<(Vec<u8>, CompiledModule)> {
+fn diem_framework_modules(release_name: &str) -> Vec<(Vec<u8>, CompiledModule)> {
     // Need to filter out Genesis module similiar to what is done in vmgenesis to make sure Genesis
     // module isn't published on-chain.
-    diem_framework_releases::current_modules_with_blobs()
+    diem_framework_releases::load_modules_from_release(release_name)
+        .unwrap_or_else(|_| {
+            panic!(
+                "Failed to load modules from given release name: {:?}",
+                release_name
+            )
+        })
         .into_iter()
-        .map(|(bytes, modules)| (bytes.clone(), modules.clone()))
+        .map(|bytes| {
+            (
+                bytes.clone(),
+                CompiledModule::deserialize(&bytes).expect("Failed to deserialize compiled module"),
+            )
+        })
+        .filter(|module| module.1.self_id().name().as_str() != GENESIS_MODULE_NAME)
```

Comment:
Do we need to worry about the Debug module or other test-only modules from the move-stdlib being picked up here?

Reply:
We do need to worry about this IMO. There isn't a way to filter out those modules effectively. From my read is debug module isn't included in move-stdlib so it won't be there.